### ISSUE TITLE
add linecard metrics (state, uptime, cpu/mem utilization)

### ISSUE THREAD

**Issue by nExXxuS** [czerwonk/junos_exporter#51]
It would be very nice to have metrics about individual FPCs. For instance there have been some PRs where memory leaks fill the fpcs memory, causing unexpected actions like sudden reboot, or crashes.

Metrics could be collected through:

* State, CPU Utilization, Memory Utilization
```
show chassis fpc
                     Temp  CPU Utilization (%)   CPU Utilization (%)  Memory    Utilization (%)
Slot State            (C)  Total  Interrupt      1min   5min   15min  DRAM (MB) Heap     Buffer
  0  Online            37     23          5       27     25     25    2048       32         33
  1  Online            37     18          5       18     19     19    3584       19         35
  2  Online            36     15          0       16     15     15    2048       32         21
```

* Uptime
```
show chassis fpc detail
Slot 0 information:
  State                               Online
  Temperature                      37 degrees C / 98 degrees F
  Total CPU DRAM                 2048 MB
  Total RLDRAM                   1036 MB
  Total DDR DRAM                 6656 MB
  Start time                          2019-02-19 23:01:59 UTC
  Uptime                              104 days, 11 hours, 8 minutes, 4 seconds
  Max power consumption            485 Watts
Slot 1 information:
  State                               Online
  Temperature                      37 degrees C / 98 degrees F
  Total CPU DRAM                 3584 MB
  Total RLDRAM                    259 MB
  Total DDR DRAM                 20352 MB
  Start time                          2019-02-19 23:02:10 UTC
  Uptime                              104 days, 11 hours, 7 minutes, 53 seconds
  Max MPC base power consumption   376 Watts
  Max MIC0 power consumption        30 Watts
  Max MIC1 power consumption        40 Watts
  Max MPC total power consumption  446 Watts
Slot 2 information:
  State                               Online
  Temperature                      36 degrees C / 96 degrees F
  Total CPU DRAM                 2048 MB
  Total RLDRAM                   1036 MB
  Total DDR DRAM                 6656 MB
  Start time                          2019-02-19 23:02:11 UTC
  Uptime                              104 days, 11 hours, 7 minutes, 52 seconds
  Max power consumption            485 Watts
```

**Comment by czerwonk** [comment]
@nExXxuS can you provide me with the output of `show chassis fpc detail | display xml`. Since I do not have access to an (v)MX at the moment. If so, I would try to implement this next weekend.

**Comment by nExXxuS** [comment]
Executed on QFX5110:
```
show chassis fpc detail | display xml
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/18.4R1/junos">
    <fpc-information xmlns="http://xml.juniper.net/junos/18.4R1/junos-chassis" junos:style="verbose">
        <fpc>
            <slot>0</slot>
            <state>Online</state>
            <temperature junos:celsius="39">39 degrees C / 102 degrees F</temperature>
            <memory-dram-size>16384</memory-dram-size>
            <memory-sram-size>0</memory-sram-size>
            <memory-sdram-size>0</memory-sdram-size>
            <start-time junos:seconds="1559804691">2019-06-06 07:04:51 UTC</start-time>
            <up-time junos:seconds="16737">4 hours, 38 minutes, 57 seconds</up-time>
        </fpc>
    </fpc-information>
    <cli>
        <banner>{master:0}</banner>
    </cli>
</rpc-reply>
```

On MX:
```
show chassis fpc detail | display xml
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/17.3R3/junos">
    <fpc-information xmlns="http://xml.juniper.net/junos/17.3R3/junos-chassis" junos:style="verbose">
        <fpc>
            <slot>1</slot>
            <state>Online</state>
            <temperature junos:celsius="30">30 degrees C / 86 degrees F</temperature>
            <memory-dram-size>2048</memory-dram-size>
            <memory-rldram-size>1036</memory-rldram-size>
            <memory-ddr-dram-size>11264</memory-ddr-dram-size>
            <start-time junos:seconds="1550101033">2019-02-13 23:37:13 UTC</start-time>
            <up-time junos:seconds="9720385">112 days, 12 hours, 6 minutes, 25 seconds</up-time>
            <max-power-consumption>584</max-power-consumption>
        </fpc>
        <fpc>
            <slot>2</slot>
            <state>Online</state>
            <temperature junos:celsius="30">30 degrees C / 86 degrees F</temperature>
            <memory-dram-size>2048</memory-dram-size>
            <memory-rldram-size>1036</memory-rldram-size>
            <memory-ddr-dram-size>11264</memory-ddr-dram-size>
            <start-time junos:seconds="1550101046">2019-02-13 23:37:26 UTC</start-time>
            <up-time junos:seconds="9720372">112 days, 12 hours, 6 minutes, 12 seconds</up-time>
            <max-power-consumption>584</max-power-consumption>
        </fpc>
        <fpc>
            <slot>9</slot>
            <state>Online</state>
            <temperature junos:celsius="28">43 degrees C / 109 degrees F</temperature>
            <memory-dram-size>3124</memory-dram-size>
            <memory-rldram-size>516</memory-rldram-size>
            <memory-ddr-dram-size>8192</memory-ddr-dram-size>
            <start-time junos:seconds="1550101152">2019-02-13 23:39:12 UTC</start-time>
            <up-time junos:seconds="9720266">112 days, 12 hours, 4 minutes, 26 seconds</up-time>
            <max-power-consumption>465</max-power-consumption>
        </fpc>
        <fpc>
            <slot>10</slot>
            <state>Online</state>
            <temperature junos:celsius="0">44 degrees C / 111 degrees F</temperature>
            <memory-dram-size>3584</memory-dram-size>
            <memory-xr2-size>259</memory-xr2-size>
            <memory-ddr-dram-size>24960</memory-ddr-dram-size>
            <start-time junos:seconds="1550101065">2019-02-13 23:37:45 UTC</start-time>
            <up-time junos:seconds="9720353">112 days, 12 hours, 5 minutes, 53 seconds</up-time>
            <max-power-consumption>496</max-power-consumption>
        </fpc>
    </fpc-information>
    <cli>
        <banner>{master}</banner>
    </cli>
</rpc-reply>
```

**Comment by czerwonk** [comment]
implemented in #53

**Comment by skorpy2009** [comment]
Output of an MX960, there should not be a difference:
```
<rpc-reply xmlns:junos="http://xml.juniper.net/junos/17.3R3/junos">
    <fpc-information xmlns="http://xml.juniper.net/junos/17.3R3/junos-chassis" junos:style="verbose">
        <fpc>
            <slot>0</slot>
            <state>Online</state>
            <temperature junos:celsius="38">38 degrees C / 100 degrees F</temperature>
            <memory-dram-size>2048</memory-dram-size>
            <memory-rldram-size>1324</memory-rldram-size>
            <memory-ddr-dram-size>5120</memory-ddr-dram-size>
            <start-time junos:seconds="1540320035">2018-10-23 18:40:35 UTC</start-time>
            <up-time junos:seconds="20179996">233 days, 13 hours, 33 minutes, 16 seconds</up-time>
            <max-power-consumption>440</max-power-consumption>
        </fpc>
        <fpc>
            <slot>1</slot>
            <state>Online</state>
            <temperature junos:celsius="37">37 degrees C / 98 degrees F</temperature>
            <memory-dram-size>2048</memory-dram-size>
            <memory-rldram-size>1324</memory-rldram-size>
            <memory-ddr-dram-size>5120</memory-ddr-dram-size>
            <start-time junos:seconds="1540320036">2018-10-23 18:40:36 UTC</start-time>
            <up-time junos:seconds="20179995">233 days, 13 hours, 33 minutes, 15 seconds</up-time>
            <max-power-consumption>440</max-power-consumption>
        </fpc>
        <fpc>
            <slot>2</slot>
            <state>Online</state>
            <temperature junos:celsius="31">31 degrees C / 87 degrees F</temperature>
            <memory-dram-size>2048</memory-dram-size>
            <memory-rldram-size>662</memory-rldram-size>
            <memory-ddr-dram-size>2560</memory-ddr-dram-size>
            <start-time junos:seconds="1540320034">2018-10-23 18:40:34 UTC</start-time>
            <up-time junos:seconds="20179997">233 days, 13 hours, 33 minutes, 17 seconds</up-time>
            <max-power-consumption>227</max-power-consumption>
        </fpc>
    </fpc-information>
    <cli>
        <banner>{master}</banner>
    </cli>
</rpc-reply>
```

**Comment by czerwonk** [comment]
yep, only memory differs between the models, so there is a memory metric with the memory tech/type as label to allow for more flexible querying

**Comment by skorpy2009** [comment]
> implemented in #53

You do not collect `CPU Utilization`

**Comment by nExXxuS** [comment]
IMHO the memery size is not very useful to collect, as its static and wont change during operation. The only useful metric of `show chassis fpc detail` is the uptime.
